src/main/java/org/jvnet/hudson/plugins/repositoryconnector/VersionParameterDefinition.java

```diff
@@ -53,35 +58,74 @@ public VersionParameterDefinition(String repoid, String groupid,
     public VersionParameterDefinition copyWithDefaultValue(ParameterValue defaultValue) {
         if (defaultValue instanceof StringParameterValue) {
             // TODO: StringParameterValue value = (StringParameterValue) defaultValue;
-            return new VersionParameterDefinition(getRepoid(), "",
-                    "", "", getDescription());
+            return new VersionParameterDefinition(getRepoid(), getGroupid(), getArtifactid(), "", getDescription(),true);
         } else {
             return this;
         }
     }
 
+    public boolean reverseOrder() {
+    	return reverseOrder;
+    }
+    
+    public Collection<Repository> getRepos(String repoid) {
+    	List<Repository> out = new ArrayList<Repository>();
+    	
+    	for(Map.Entry<String,Repository> e : RepositoryConfiguration.get().getRepositoryMap().entrySet()) {
+    		if("ALL".equals(repoid) || repoid.equals(e.getKey())) {
```

Comment:
This causes esisting jobs to fails with this exception:
```
Caused by: java.lang.NullPointerException
	at org.jvnet.hudson.plugins.repositoryconnector.VersionParameterDefinition.getRepos(VersionParameterDefinition.java:74)
	at org.jvnet.hudson.plugins.repositoryconnector.VersionParameterDefinition.getChoices(VersionParameterDefinition.java:85)
```
you must make sure, that all new parameters have a resonable default, which should lead to the old behavior. In this case before you introduced `repoid`, so you you should set `repoid` to `ALL` as a default.